packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -178,17 +183,20 @@ export abstract class FunctionBase extends Resource implements IFunction {
   /**
    * Adds a permission to the Lambda resource policy.
    * @param id The id ƒor the permission construct
+   * @param permission The permission to grant to this Lambda function. @see Permission for details.
+   * @param scope An optional scope to which this permission should be attached. By default, this will be the function handler.
    */
-  public addPermission(id: string, permission: Permission) {
+  public addPermission(id: string, permission: Permission, scope?: cdk.Construct) {
```

Comment:
Feels like this should be added to `Permission` and not as a 3rd argument to `addPermission`

Reply:
Is that ok?

I wasn't sure if I can push information unrelated to modeling the 'permission' itself into this structure. I chose the option similar to the constructors where the scope and id are outside of the props.
Thoughts?